Finalize 8.3 release notes (#25794)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

The Gradle team is excited to announce Gradle @version@.

This release features the support for [persistent Java compiler daemons](#faster-java-compilation) to speed up Java compilation.
Gradle will also use [less memory for dependency resolution](#reduced-memory-consumption).

> The effect is significant for large builds, in particular, large Android builds including, but not limited to, IDE sync.

Reviewer: ```suggestion
The effect is significant, particularly for large builds such as Android builds.```

> The effect is significant for large builds, in particular, large Android builds including, but not limited to, IDE sync.

Reviewer: ```suggestion
The effect is significant, particularly for large builds such as Android builds.
```

Reviewer: I don't see a way to add "IDE sync". It's just a random add at this point.

Author: Thanks, we have the specifics in the section. Making it shorter for the preamble makes sense IMO

The embedded Kotlin has been updated from 1.8.20 to [Kotlin 1.9.0](https://github.com/JetBrains/kotlin/releases/tag/v1.9.0), unblocking the support for running on Java 20.

The Kotlin language and API levels for the Kotlin DSL are still set to 1.8 for backward compatibility.
See the release notes for [Kotlin 1.8.22](https://github.com/JetBrains/kotlin/releases/tag/v1.8.22) and [Kotlin 1.8.21](https://github.com/JetBrains/kotlin/releases/tag/v1.8.21).

### Configuration cache

The [configuration cache](userguide/configuration_cache.html) improves build time by caching the result of the configuration phase and reusing it for subsequent builds. This feature can significantly improve build performance.

#### Temporarily opting out of stricter configuration inputs detection

In order to ensure correctness, configuration cache detects undeclared inputs when plugins or build logic perform file system checks such as `File.exists()` or `File.isFile()`.
However, not all plugins might be updated yet to properly declare all the inputs, leading to cache misses and suboptimal performance.

Configuration cache now has an opt-out option for temporarily ignoring such undeclared inputs.

> This option mitigates undesired configuration cache misses, until the plugins are properly updated.

Reviewer: ```suggestion
This option mitigates undesired configuration cache misses until the plugins are properly updated.
```
